web/src/components/RunsTable/runsTable.js

```diff
@@ -181,247 +181,267 @@ class RunsTable extends Component {
-          sort: '-_id',
-          query: queryString
-        }
-      }),
-      axios.get('/api/v1/Runs', {
-        params: {
-          distinct: 'omniboard.tags'
-        }
-      }),
-      axios.get('/api/v1/Omniboard.Columns'),
-      axios.get('/api/v1/Omniboard.Config.Columns')
-    ])
-    .then(axios.spread((runsResponse, tags, metricColumns, configColumns) => {
-      let runsResponseData = runsResponse.data;
+
+    // First retrieve metric columns as to decide which metrics need to be populated.
+    axios.get('/api/v1/Omniboard.Columns').then((metricColumns) => {
       const metricColumnsData = metricColumns.data;
-      const configColumnsData = configColumns.data;
-      if (runsResponseData && runsResponseData.length) {
-        const _defaultSortIndices = [];
-        runsResponseData = runsResponseData.map(data => {
-          if ('config' in data) {
-            const config = data['config'];
-            // Expand each key of config column as individual columns
-            delete data['config'];
-            data = {...data, ...config};
-            const configNameMap = this._getColumnNameMap(config, 'config');
-            columnNameMap = {...columnNameMap, ...configNameMap};
-
-            // Include config columns
-            if (configColumnsData.length) {
-              const configColumnsObject = {};
-              const configColumnNameMap = {};
-              configColumnsData.forEach(column => {
-                const columnName = column.name;
-                const configPath = column.config_path;
-                configColumnsObject[columnName] = this._resolveObjectPath(config, configPath, '');
-                configColumnNameMap[columnName] = `config.${configPath}`;
-              });
-              data = {...data, ...configColumnsObject};
-              columnNameMap = {...columnNameMap, ...configColumnNameMap};
-            }
-          }
 
-          if ('experiment' in data) {
-            const experiment = data['experiment'];
-            delete data['experiment'];
-            data = {...data, 'experiment_name': experiment['name']};
-            columnNameMap = {...columnNameMap, 'experiment_name': 'experiment.name'};
-          }
+      const run_query_params  ={
+        select: '_id,heartbeat,experiment,command,artifacts,host,stop_time,config,' +
+                'result,start_time,resources,format,status,omniboard,metrics,meta',
+        sort: '-_id',
+        query: queryString
+      };
 
-          if ('host' in data) {
-            const host = data['host'];
-            delete data['host'];
-            data = {...data, 'hostname': host['hostname']};
-            const hostMap = this._getColumnNameMap(host, 'host');
-            columnNameMap = {...columnNameMap, ...hostMap};
+      if (metricColumnsData.length){
+        const metric_column_names = [];
+        metricColumnsData.forEach(column => {
+          metric_column_names.push(column.metric_name);
+        });
```

Comment:
This block can be replaced with
```
const metricColumnNames = metricColumnsData.map(column => column.metric_name);
```

Reply:
I changed the code style. But I currently have no Idea how to change the tests in runsTable.test.js ... (Never worked with mockAxios).

Reply:
Let me try to help you with that when I find time. I will push commits to your branch - matzek92:fix-memory-issue